tests/unit/asyn/wallet/test_main.py

```diff
@@ -0,0 +1,358 @@
+        )
+
+        self.assertEqual(wallet.public_key, seed_constants["ed25519_public_key"])
+        self.assertEqual(wallet.private_key, seed_constants["ed25519_private_key"])
+        self.assertEqual(
+            wallet.classic_address, seed_constants["ed25519_classic_address"]
+        )
+
+    def test_from_secret_using_regular_key_pair_using_no_algorithm(self):
+        wallet = Wallet.from_secret(
+            regular_key_pair_constants["seed"],
+            master_address=regular_key_pair_constants["master_address"],
+        )
+
+        self.assertEqual(
+            wallet.public_key, regular_key_pair_constants["ed25519_public_key"]
+        )
+        self.assertEqual(
+            wallet.private_key, regular_key_pair_constants["ed25519_private_key"]
+        )
+        self.assertEqual(
+            wallet.classic_address, regular_key_pair_constants["master_address"]
+        )
+
+    def test_from_secret_using_regular_key_pair_using_algorithm_ed25519(self):
+        wallet = Wallet.from_secret(
+            regular_key_pair_constants["seed"],
+            master_address=regular_key_pair_constants["master_address"],
+            algorithm=CryptoAlgorithm.ED25519,
+        )
+
+        self.assertEqual(
+            wallet.public_key, regular_key_pair_constants["ed25519_public_key"]
+        )
+        self.assertEqual(
+            wallet.private_key, regular_key_pair_constants["ed25519_private_key"]
+        )
+        self.assertEqual(
+            wallet.classic_address, regular_key_pair_constants["master_address"]
+        )
+
+    def test_from_secret_using_regular_key_pair_using_algorithm_edcsa_secp256k1(self):
+        wallet = Wallet.from_secret(
+            regular_key_pair_constants["seed"],
+            master_address=regular_key_pair_constants["master_address"],
+            algorithm=CryptoAlgorithm.SECP256K1,
+        )
+
+        self.assertEqual(
+            wallet.public_key, regular_key_pair_constants["secp256k1_public_key"]
+        )
+        self.assertEqual(
+            wallet.private_key, regular_key_pair_constants["secp256k1_private_key"]
+        )
+        self.assertEqual(
+            wallet.classic_address, regular_key_pair_constants["master_address"]
+        )
+
+    def test_from_entropy_using_default_algorithm(self):
+        wallet = Wallet.from_entropy(entropy_constants["entropy"])
+
+        self.assertEqual(wallet.public_key, entropy_constants["ed25519_public_key"])
+        self.assertEqual(wallet.private_key, entropy_constants["ed25519_private_key"])
+
+    def test_from_entropy_using_algorithm_ecdsa_secp256k1(self):
+        wallet = Wallet.from_entropy(
+            entropy_constants["entropy"], algorithm=CryptoAlgorithm.SECP256K1
+        )
+
+        self.assertEqual(wallet.public_key, entropy_constants["secp256k1_public_key"])
+        self.assertEqual(wallet.private_key, entropy_constants["secp256k1_private_key"])
+
+    def test_from_entropy_using_algorithm_ed25519(self):
```

Comment:
You can probably simplify these tests by splitting out a helper function which does the checks with parameters like `algorithm` + `constants` 

(Instead of `entropy_constants["secp256k1_public_key"]` you could organize it like `entropy_constants["secp256k1"]["public_key"]` which would make it easy to scalably dereference with those two parameters :) )

Reply:
General pattern: If you're specifying two values in a name, may be better to have those both be indices